ts/features/fci/screens/valid/FciDocumentsScreen.tsx

```diff
@@ -23,16 +38,195 @@ const styles = StyleSheet.create({
+      await pdfDoc.then(res => {
+        pipe(
+          res.findPageForAnnotationRef(
+            res.getForm().getSignature(uniqueName).ref
+          ),
+          O.fromNullable,
+          O.map(pageRef => {
+            const page = res.getPages().indexOf(pageRef);
+            setSignaturePage(page + 1);
+            // The signature field is extracted by its unique_name.
+            // Using low-level acrofield (acrobat field) it is possible
+            // to obtain the elements of the signature field such as the
+            // box that contains it. Once the box is obtained, its
+            // coordinates are used to draw a rectangle on the related page.
+            const signature = res.getForm().getSignature(uniqueName);
+            const [widget] = signature.acroField.getWidgets();
+            const rect = widget.getRectangle();
+            res.getPage(page).drawRectangle({
+              x: rect.x,
+              y: rect.y,
+              width: rect.width,
+              height: rect.height,
+              color: rgb(0, 0.77, 0.79),
+              opacity: 0.5,
+              borderOpacity: 0.75
+            });
+          })
+        );
+
+        return res
+          .saveAsBase64()
+          .then(r => setPdfString(`data:application/pdf;base64,${r}`));
+      });
+    },
+    [documents, currentDoc]
+  );
+
+  /**
+   * function to draw a rect over pdf giving a coordinates
+   * @param attrs the signature field attrs containing the coords
+   */
+  const drawRectangleOverSignatureFieldByCoordinates = React.useCallback(
+    async (attrs: SignatureFieldToBeCreatedAttrs) => {
+      // TODO: refactor this function to use fp-ts
+      const doc = documents[currentDoc];
+      const url = doc.url;
+
+      const existingPdfBytes = await ReactNativeBlobUtil.fetch("GET", url).then(
+        res => res.base64()
+      );
+
+      const pdfDoc = PDFDocument.load(
+        `data:application/pdf;base64,${existingPdfBytes}`
+      );
+
+      await pdfDoc.then(res => {
+        const page = attrs.page;
+        setSignaturePage(page);
+        // The signature box is drawn using the coordinates of the signature field.
+        res.getPage(page).drawRectangle({
+          x: attrs.coordinates.top_right.x,
+          y: attrs.coordinates.top_right.y,
+          width: attrs.coordinates.bottom_left.x,
+          height: attrs.coordinates.bottom_left.y,
+          color: rgb(0, 0.77, 0.79),
+          opacity: 0.5,
+          borderOpacity: 0.75
+        });
+
+        return res
+          .saveAsBase64()
+          .then(r => setPdfString(`data:application/pdf;base64,${r}`));
+      });
+    },
+    [documents, currentDoc]
+  );
+
+  const onSignatureDetail = React.useCallback(
+    (attrs: ExistingSignatureFieldAttrs | SignatureFieldToBeCreatedAttrs) => {
+      if (hasUniqueName(attrs)) {
+        drawRectangleOverSignatureFieldById(attrs.unique_name).catch(
+          readableReport
```

Comment:
what happens visually in case of errors here?

Reply:
In every case (with unique_name, coords or empty attrs) visually it should be displayed the PDF without rect. I think it should somehow alert the user visually but it hasn't been represented as of yet in UI design. It could be a future refinement.